File: Transactions.java
import java.util.ArrayList;
import java.util.Locale;
import java.util.Scanner;

public class Transactions {

    public void operationMenu(ArrayList<Employee> employeeList) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Ingrese su nombre");
        String nameToCheck = sc.nextLine().toLowerCase(Locale.ROOT);
        System.out.println("Ingrese su apellido");
        String lastnameToCheck = sc.nextLine().toLowerCase();

        int ID = 0;
        int identifier = -1;
        for (Employee emple : employeeList) {

            if (emple.getName().toLowerCase().equals(nameToCheck) && emple.getLastname().toLowerCase().equals(lastnameToCheck)) {
                identifier = ID;
                break;
            }
            ID++;

        }

        if (identifier > -1) {
            Employee employeeSel = employeeList.get(identifier);
            int ind = 0;

            System.out.print(">>>Bienvenido usuario " + employeeSel.getName() + " " + employeeSel.getLastname() + " :D \nCuentas : " + "\n");
            int ID2 = 0;
            for (BankAccount bankAccount : employeeSel.getListAccounts()) {

                System.out.println("§ ID :" + (ID2 + 1) + " | Cuenta: " + employeeSel.getListAccounts().get(ind).getAccountNumber() + " |  Tipo: " + employeeSel.getListAccounts().get(ind).getType() + " | Cantidad: " + employeeSel.getListAccounts().get(ind).getAmount() + " §");
                ind++;
                ID2++;
            }
            System.out.println("Seleccione una cuenta (ID)");
            int accountIndex = sc.nextInt();
            System.out.println("Seleccionaste: \n" +
                    employeeSel.getListAccounts().get(accountIndex - 1).getAccountNumber() +
                    "\t. Tipo: " + employeeSel.getListAccounts().get(accountIndex - 1).getType());

            //quiero seleccionar una cuenta y despues poner un menu para ver si desea agregar o retirar dinero
            boolean band = true;
            do {
                System.out.println("\t<<<Operaciones>>>\n1) Depositar dinero " + "\n2) Retirar dinero\n3 Consultar saldo\n0)Regresar\n");
                int option = sc.nextInt();
                switch (option) {
                    case 1 -> {
                        System.out.println("¿Cuánto desea depositar?");
                        double amount = sc.nextDouble();
                        employeeSel.getListAccounts().get(accountIndex - 1).depositMoney(amount);
                    }
                    case 2 -> {
                        System.out.println("¿Cuánto desea retirar?");
                        double amount = sc.nextDouble();
                        employeeSel.getListAccounts().get(accountIndex - 1).withdrawMoney(amount);
                    }
                    case 3 -> {
                        System.out.println("Su saldo es de: $ " + employeeSel.getListAccounts().get(accountIndex - 1).getAmount());
                    }
                    case 0 -> {
                        band = false;
                    }
                }
                sc.nextLine();
            } while (band);
        } else {
            System.out.println("Usuario invalido :o");
        }
    }


}

File: transactions.py


class Transactions:
    def operation_menu(self, employee_list):
        print("Ingrese su nombre")
        name_to_check = input().lower()
        print("Ingrese su apellido")
        lastname_to_check = input().lower()

        identifier = -1
        for id_, employee in enumerate(employee_list):
            if employee.get_name().lower() == name_to_check and employee.get_last_name().lower() == lastname_to_check:
                identifier = id_
                break

        if identifier > -1:
            employee_sel = employee_list[identifier]
            ind = 0

            print(">>>Bienvenido usuario " + employee_sel.get_name() + " " + employee_sel.get_last_name() +
                  " :D \nCuentas : " + "\n")
            id_2 = 0
            for _ in employee_sel.get_list_accounts():
                print("§ ID :" + str(id_2 + 1) + " | Cuenta: " + str(employee_sel.get_list_accounts()[ind].get_account_number()) + " |  Tipo: " + str(employee_sel.get_list_accounts()[ind].get_account_type()) + " | Cantidad: " + str(employee_sel.get_list_accounts()[ind].get_amount()) + " §")
                ind += 1
                id_2 += 1
            print("Seleccione una cuenta (ID)")
            account_index = int(input())
            print("Seleccionaste: \n" +
                  str(employee_sel.get_list_accounts()[account_index - 1].get_account_number()) +
                  "\t. Tipo: " + str(employee_sel.get_list_accounts()[account_index - 1].get_account_type()))

            # quiero seleccionar una cuenta y después poner un menú para ver si desea agregar o retirar dinero
            band = True
            while band:
                print("\t<<<Operaciones>>>\n1) Depositar dinero " + "\n2) Retirar dinero\n3 Consultar saldo\n0)Regresar\n")
                option = int(input())
                if option == 1:
                    print("¿Cuánto desea depositar?")
                    amount = float(input())
                    employee_sel.get_list_accounts()[account_index - 1].deposit_money(amount)
                elif option == 2:
                    print("¿Cuánto desea retirar?")
                    amount = float(input())
                    employee_sel.get_list_accounts()[account_index - 1].withdraw_money(amount)
                elif option == 3:
                    print("Su saldo es de: $ " + str(employee_sel.get_list_accounts()[account_index - 1].get_amount()))
                elif option == 0:
                    band = False
        else:
            print("Usuario invalido :o")
